### PR TITLE
Add warning for empty requirements files

### PR DESCRIPTION
**Pull request by zanieb**
Also, improve tracing of requirements file parsing.

Per my confusion in #1334

### CODE REVIEW

File: crates/uv/src/requirements.rs [hunk 1]
```diff
@@ -79,6 +80,18 @@ impl RequirementsSource {
     }
 }
 
+impl std::fmt::Display for RequirementsSource {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Editable(path) => write!(f, "-e {path}"),
+            Self::RequirementsTxt(path) | Self::PyprojectToml(path) => {
+                write!(f, "{}", path.to_string_lossy())
```
Comment (charliermarsh) [inline crates/uv/src/requirements.rs:88]: `path.display()`? Or what type is this?

**Review by charliermarsh — APPROVED** [review]

File: crates/uv/src/requirements.rs [hunk 2]
```diff
@@ -79,6 +80,18 @@ impl RequirementsSource {
     }
 }
 
+impl std::fmt::Display for RequirementsSource {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Editable(path) => write!(f, "-e {path}"),
+            Self::RequirementsTxt(path) | Self::PyprojectToml(path) => {
+                write!(f, "{}", path.display())
```
Comment (charliermarsh) [inline crates/uv/src/requirements.rs:88]: Oops, it should actually be `.normalized_display`

Comment (zanieb) [reply crates/uv/src/requirements.rs:88]: Ugh lol

Comment (zanieb) [reply crates/uv/src/requirements.rs:88]: What's different? Windows?

**Pull request merged**